R/helpers.R

```diff
@@ -0,0 +1,40 @@
+#' @title Convert epidemic model output to a data.frame
+#'
+#' @param l A list, usually the output of a call to [.epidemic_default_cpp()].
+#' @param compartments A character vector of the compartment names.
+#' @return A data.frame with N+1 columns, where N is the number of compartments,
+#' with the additional column being "time".
+#' @export
+output_to_df <- function(l, compartments = c(
+                           "susceptible", "exposed",
+                           "infectious", "recovered",
+                           "vaccinated"
+                         )) {
+  # input checking
+  stopifnot(
+    "Error: `l` must be a list" =
+      is.list(l),
+    "Error: `l` must have names 'x' and 'time'" =
+      names(l) %in% c("x", "time")
```

Comment:
```suggestion
      all(c("x", "time") %in% names(l))
```

I'm not completely sure since I don't have an example to test it but it sounds like you want to check if `l` contains the names `x` *and* `time`.
At the moment, you are checking that it contains *only* `x` *or* `time`.

You also need an `all` or `any` to pass a logical of length 1

Reply:
Yes, I expect only "x" and "time" for now - this function will probably undergo substantial change as the output from Rcpp is better worked out. Will add the `all()` for now.

Reply:
Added `all()` in https://github.com/epiverse-trace/epidemics/commit/7b5c7034cb97b4a48c07fbc1b79bd16bd59265bf